### ISSUE TITLE
Support for CorrelationCost1D

### ISSUE THREAD

**Issue by serizba** [tensorflow/addons#1907]
**Describe the feature and the current behavior/state.**
Support for CorrelationCost along just 1 dimension. Currently CorrelationCost applies the kernel in both directions, horizontal and vertical, as described in the original FlowNet paper. When working with rectified stereo images, you may only need to compute the correlation cost in horizontal lines, as done in [DispNet](https://lmb.informatik.uni-freiburg.de/Publications/2016/MIFDB16/paper-MIFDB16.pdf).

**Relevant information**
- Are you willing to contribute it (yes/no): No
- Are you willing to maintain it going forward? (yes/no): No
- Is there a relevant academic paper? (if so, where): [DispNet](https://lmb.informatik.uni-freiburg.de/Publications/2016/MIFDB16/paper-MIFDB16.pdf)
- Is there already an implementation in another framework? (if so, where): [lmbspecialops repository](https://github.com/lmb-freiburg/lmbspecialops/tree/master/src) Repository with different operations used in FlowNet, DispNet and others.
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
It should be a layer as CorrelationCost
**Who will benefit with this feature?**
Researchers working with rectified stereo images where the information should only be searched along horizontal lines.

**Comment by chekoduadarsh** [comment]
@seanpmorgan 

Hey, I would like to implement this feature. (This is my first time contributing to Tensorflow) Could you assign this to me?

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
